collector/collector.go

```diff
@@ -0,0 +1,360 @@
+		State: []string{"scheduled", "running"},
+		ListOptions: bk.ListOptions{
+			PerPage: recordsPerPage,
+		},
+	})
+
+	return currentBuilds.Pages(func(v interface{}) bool {
+		for _, build := range v.([]bk.Build) {
+			if c.Opts.Debug {
+				log.Printf("Adding build to stats (id=%q, pipeline=%q, branch=%q, state=%q)",
+					*build.ID, *build.Pipeline.Name, *build.Branch, *build.State)
+			}
+
+			pipeline := *build.Pipeline.Name
+
+			if _, ok := r.Pipelines[pipeline]; !ok {
+				r.Pipelines[pipeline] = newCounts()
+			}
+
+			switch *build.State {
+			case "running":
+				r.Totals[RunningBuildsCount]++
+				r.Pipelines[pipeline][RunningBuildsCount]++
+
+			case "scheduled":
+				r.Totals[ScheduledBuildsCount]++
+				r.Pipelines[pipeline][ScheduledBuildsCount]++
+			}
+
+			var buildQueues = map[string]int{}
+
+			for _, job := range build.Jobs {
+				if job.Type != nil && *job.Type == "waiter" {
+					continue
+				}
+
+				state := ""
+				if job.State != nil {
+					state = *job.State
+				}
+
+				if c.Opts.Debug {
+					log.Printf("Adding job to stats (id=%q, pipeline=%q, queue=%q, type=%q, state=%q)",
+						*job.ID, *build.Pipeline.Name, queue(job), *job.Type, state)
+				}
+
+				if _, ok := r.Queues[queue(job)]; !ok {
+					r.Queues[queue(job)] = newCounts()
+				}
+
+				if state == "running" || state == "scheduled" {
+					switch state {
+					case "running":
+						r.Totals[RunningJobsCount]++
+						r.Queues[queue(job)][RunningJobsCount]++
+						r.Pipelines[pipeline][RunningJobsCount]++
+
+					case "scheduled":
+						r.Totals[ScheduledJobsCount]++
+						r.Queues[queue(job)][ScheduledJobsCount]++
+						r.Pipelines[pipeline][ScheduledJobsCount]++
+					}
+
+					r.Totals[UnfinishedJobsCount]++
+					r.Queues[queue(job)][UnfinishedJobsCount]++
+					r.Pipelines[pipeline][UnfinishedJobsCount]++
+				}
+
+				buildQueues[queue(job)]++
+			}
+
+			// add build metrics to queues
+			if len(buildQueues) > 0 {
+				for queue := range buildQueues {
+					switch *build.State {
+					case "running":
+						r.Queues[queue][RunningBuildsCount]++
+
+					case "scheduled":
+						r.Queues[queue][ScheduledBuildsCount]++
+					}
+				}
+			}
+		}
+		return true
+	})
+}
+
+func (c *Collector) addAgentMetrics(r *Result) error {
+	p := &pager{
+		lister: func(page int) (interface{}, int, error) {
+			agents, resp, err := c.agentService.List(c.Opts.OrgSlug, &bk.AgentListOptions{
+				ListOptions: bk.ListOptions{
+					Page: page,
+				},
+			})
+			if err != nil {
+				return nil, 0, err
+			}
+			return agents, resp.NextPage, err
+		},
+	}
+
+	r.Totals[BusyAgentCount] = 0
+	r.Totals[IdleAgentCount] = 0
+	r.Totals[TotalAgentCount] = 0
+
+	for queue := range r.Queues {
+		r.Queues[queue][BusyAgentCount] = 0
+		r.Queues[queue][IdleAgentCount] = 0
+		r.Queues[queue][TotalAgentCount] = 0
+	}
+
+	err := p.Pages(func(v interface{}) bool {
+		agents := v.([]bk.Agent)
+
+		for _, agent := range agents {
+			queue := "default"
```

Comment:
ORLY? That's interesting. I will keep this behaviour in currently and look to fix it in a subsequent PR.